src/mysql/MySqlConnection.cc

```diff
@@ -67,130 +65,126 @@ struct InitializeMysqlLibrary {
         janitor.reset(new MySqlThreadJanitor);
     }
 };
-}  // namespace
 
-namespace lsst::qserv::mysql {
+/**
+ * Establish a new MySQL connection.
+ * @param config Parameters of the connection.
+ * @return A pointer to the MySQL connection or nullptr.
+ */
+MYSQL* doConnect(std::shared_ptr<lsst::qserv::mysql::MySqlConfig> const& config) {
+    // We must call mysql_library_init() exactly once before calling mysql_init
+    // because it is not thread safe. Both mysql_library_init and mysql_init
+    // call mysql_thread_init, and so we must arrange to call mysql_thread_end
+    // when the calling thread exists. We do this by allocating a thread
+    // local object that calls mysql_thread_end from its destructor.
+    static std::once_flag initialized;
+    static thread_local std::unique_ptr<MySqlThreadJanitor> janitor;
 
-MySqlConnection::MySqlConnection()
-        : _mysql(nullptr),
-          _mysql_res(nullptr),
-          _isConnected(false),
-          _isExecuting(false),
-          _interrupted(false) {}
-
-MySqlConnection::MySqlConnection(MySqlConfig const& sqlConfig)
-        : _mysql(nullptr),
-          _mysql_res(nullptr),
-          _isConnected(false),
-          _sqlConfig(std::make_shared<MySqlConfig>(sqlConfig)),
-          _isExecuting(false),
-          _interrupted(false) {}
-
-MySqlConnection::~MySqlConnection() {
-    if (_mysql) {
-        if (_mysql_res) {
-            MYSQL_ROW row;
-            while ((row = mysql_fetch_row(_mysql_res)))
-                ;  // Drain results.
-            _mysql_res = nullptr;
-        }
-        closeMySqlConn();
+    std::call_once(initialized, InitializeMysqlLibrary(janitor));
+    MYSQL* m = mysql_init(nullptr);
+    if (nullptr == m) return m;
+    if (nullptr == janitor) janitor.reset(new MySqlThreadJanitor);
+    unsigned long const clientFlag = CLIENT_MULTI_STATEMENTS;
+    mysql_options(m, MYSQL_OPT_LOCAL_INFILE, 0);
+    MYSQL* c = mysql_real_connect(m, config->socket.empty() ? config->hostname.c_str() : 0,
+                                  config->username.empty() ? 0 : config->username.c_str(),
+                                  config->password.empty() ? 0 : config->password.c_str(),
+                                  config->dbName.empty() ? 0 : config->dbName.c_str(), config->port,
+                                  config->socket.empty() ? 0 : config->socket.c_str(), clientFlag);
+    if (nullptr == c) {
+        // Failed to connect: free resources.
+        mysql_close(m);
+        return c;
     }
+    return m;
```

Comment:
I'd like to avoid single letter variables. Trying to highlight or find them results in lots of false positives.

Reply:
100% agree with you! This code (and the variable) already existed. So, I decided to leave it as is for now. The connector and Qserv MySQL API classes soon will be redesigned and reimplemented.